### PR TITLE
Syscalls topic

### PR DESCRIPTION
**Pull request by bverschueren**

### CODE REVIEW

**Comment by nibalizer** [comment]
This looks good, though it could use a squash.

File: syscalls.rst [hunk 1]
```diff
@@ -3,23 +3,188 @@ Syscalls
...
+	<..>
+	read(3, "#!/bin/bash\necho \"one line of te"..., 80) = 80
+	<..>
+	dup2(3, 255)                            = 255
+	close(3)                                = 0
+	<..>
+	read(255, "#!/bin/bash\necho \"one line of te"..., 126) = 126
+	<..>
+
+``open`` prepares the file for reading in file descriptor 3 (result of the syscall ``open``). Next, ``read`` attempts to read the next 80 bytes from fd (file descriptor) 3. At this point the interpreter line of the script is read, indicating that the /bin/bash binary needs to be invoked to process the script.
```
Comment (miketheman) [inline syscalls.rst]: In this paragraph, and other lines, split new sentences to new lines.

**Comment by miketheman** [comment]
Indeed a good start. I'd update the style for newlines, and quash commits.

**Comment by bverschueren** [comment]
OK, I've split the sentences into newlines and squashed the intermediate commits

**Comment by miketheman** [comment]
@bverschueren I think you might have not merged/pushed something correctly after rebase/squash - I now see 11 commits for this change.

**Comment by bverschueren** [comment]
@miketheman yes, had to force the push to update the pull request

**Comment by miketheman** [comment]
@bverschueren awesome.

One last bit is that these commits were done with an invalid email address: `From: bverschueren <bverschueren>`

You might consider updating your git config to properly set this email address to a valid one associated with your GitHub account, amend and re-force-push.

**Comment by bverschueren** [comment]
@miketheman Ok, done

**Comment by miketheman** [comment]
Awesome, thanks!

**Pull request merged**